backend/fms_core/services/container.py

```diff
@@ -129,10 +129,10 @@ def get_or_create_container(barcode, kind=None, name=None, coordinates=None, con
                 errors.append(f"Provided container kind {kind} does not match the container kind {container.kind} of the container retrieved using the barcode {barcode}.")
             if name and name != container.name:
                 errors.append(f"Provided container name {name} does not match the container name {container.name} of the container retrieved using the barcode {barcode}.")
-            if container_parent and container_parent.id != container.location.id:
-                errors.append(f"Provided parent container {container_parent.barcode} does not match the parent container {container.location.barcode} of the container retrieved using the barcode {barcode}.")
+            if container_parent and container_parent.id != container.location_id:
+                errors.append(f"Provided parent container {container_parent.barcode} does not match the parent container ({container.location.barcode if container.location is not None else 'None'}) of the container retrieved using the barcode {barcode}.")
```

Comment:
seeing None as the parent container might be a little confusing for users, but i don't mind